### PR TITLE
Document locking and unlocking for MQTT lock

### PR DESCRIPTION
**Pull request by Nardol**
## Proposed change

Document `state_locking` and `state_unlocking` in the MQTT lock documentation

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#85779
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/lock.mqtt.markdown [hunk 1]
```diff
@@ -20,6 +20,8 @@ Optimistic mode can be forced, even if state topic is available. Try to enable i
 
 It's mandatory for locks to support `lock` and `unlock`. A lock may optionally support `open`, (e.g. to open the bolt in addition to the latch), in this case, `payload_open` is required in the configuration. If the lock is in optimistic mode, it will change states to `unlocked` when handling the `open` command.
 
+`unlocking`, `locking` and `jammed` are also supported.
+
```
Comment (jbouwh) [inline source/_integrations/lock.mqtt.markdown]:
```suggestion

An MQTT lock can also report the intermediate states `unlocking`, `locking` and a state when the lock is `jammed`.
```

**Comment by jbouwh** [comment]
Please update also the abbreviations on the MQTT integration page.

File: source/_integrations/mqtt.markdown [hunk 2]
```diff
@@ -431,8 +431,11 @@ Configuration variable names in the discovery payload may be abbreviated to cons
     'stat_open':           'state_open',
     'stat_opening':        'state_opening',
     'stat_stopped':        'state_stopped',
+    'stat_jammed':         'state_jammed',
```
Comment (jbouwh) [inline source/_integrations/mqtt.markdown]:
```suggestion
```

**Review by jbouwh — CHANGES_REQUESTED** [review]
One leftover of `jammed` state.

**Comment by Nardol** [comment]
> One leftover of `jammed` state.

Did I forget another than this I commit at last?
I hop I won't forget abbreviations for next PR.

**Review by jbouwh — APPROVED** [review]
Looks good!

**Pull request merged**